torch_geometric/utils/_scatter.py

```diff
@@ -65,6 +65,8 @@ def scatter(
         size = src.size()[:dim] + (dim_size, ) + src.size()[dim + 1:]
 
         # For "any" reduction, we use regular `scatter_`:
+        if index.dtype != torch.int64:
+            index = index.to(torch.int64)
```

Comment:
Can you clarify for me which `edge_index` is not of `dtype=torch.int64`? I think we should fix the root cause instead of silently casting to `int64`.

Reply:
Saw your fix commit, thanks!